src/http/http_server.cpp

```diff
@@ -178,25 +178,41 @@ void http_server::add_service(http_service *service)
     return ret;
 }
 
-message_ptr http_response::to_message(message_ex *req) const
+/*extern*/ void http_response_reply(const http_response &resp, message_ex *req)
 {
-    message_ptr resp = req->create_response();
+    message_ptr resp_msg = req->create_response();
 
     std::ostringstream os;
-    os << "HTTP/1.1 " << http_status_code_to_string(status_code) << "\r\n";
-    os << "Content-Type: " << content_type << "\r\n";
-    os << "Content-Length: " << body.length() << "\r\n";
-    if (!location.empty()) {
-        os << "Location: " << location << "\r\n";
+    os << "HTTP/1.1 " << http_status_code_to_string(resp.status_code) << "\r\n";
+    os << "Content-Type: " << resp.content_type << "\r\n";
+    os << "Content-Length: " << resp.body.length() << "\r\n";
+    if (!resp.location.empty()) {
+        os << "Location: " << resp.location << "\r\n";
     }
     os << "\r\n";
-    os << body;
+    os << resp.body;
 
-    rpc_write_stream writer(resp.get());
+    rpc_write_stream writer(resp_msg.get());
     writer.write(os.str().data(), os.str().length());
     writer.flush();
 
-    return resp;
+    dsn_rpc_reply(resp_msg.get());
+}
+
+/*extern*/ void start_http_server()
+{
+    // starts http server as a singleton
+    static http_server server;
```

Comment:
After http_server object constructed, then it can serve request immediatly? without calling any functions like 'start_work'?

Reply:
No. There's no such `start_work` thing. The http_server actually does not manage the TCP conns. It only registers the RPC code to the underlying RPC manager.